src/plugins/vis_type_timeline/server/series_functions/graphite.js

```diff
@@ -35,6 +35,86 @@ import _ from 'lodash';
 import fetch from 'node-fetch';
 import moment from 'moment';
 import Datasource from '../lib/classes/datasource';
+import dns from 'dns-sync';
+import IPCIDR from 'ip-cidr';
+
+const MISS_CHECKLIST_MESSAGE = `Please configure on the opensearch_dashbpards.yml file. 
+You can always enable the default allowlist configuration.`;
+
+const INVALID_URL_MESSAGE = `The Graphite URL provided by you is invalid. 
+Please update your config from OpenSearch Dashboards's Advanced Setting.`;
+
+/**
+ * Resolve hostname to IP address
+ * @param {object} urlObject
+ * @returns {string} configuredIP
+ * or null if it cannot be resolve
+ * According to RFC, all IPv6 IP address needs to be in []
+ * such as [::1]
+ * So if we detect a IPv6 address, we remove brackets
+ */
+function getIpAddress(urlObject) {
+  const hostname = urlObject.hostname;
+  const configuredIP = dns.resolve(hostname);
+  if (configuredIP) {
+    return configuredIP;
+  }
+  if (hostname.startsWith('[') && hostname.endsWith(']')) {
+    return hostname.substr(1).slice(0, -1);
+  }
+  return null;
+}
+
+function isValidURL(configuredUrl, blockedUrls) {
+  // Check the format of URL, URL has be in the format as
+  // scheme://server/path/resource otherwise an TypeError
+  // would be thrown
+  let configuredUrlObject;
+  try {
+    configuredUrlObject = new URL(configuredUrl);
+  } catch (err) {
+    return false;
+  }
+
+  const ip = getIpAddress(configuredUrlObject);
+  if (!ip) {
+    return false;
+  }
+
+  // IPCIDR check if a specific IP address fall in the
+  // range of an IP address block
+  // @param {string} bl
+  // @returns {object} cidr
+  for (const bl of blockedUrls) {
+    const cidr = new IPCIDR(bl);
+    if (cidr.contains(ip)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Check configured url using blocklist and allowlist
+ * If allowlist is used, return false if allowlist does not contain configured url
+ * If blocklist is used, return false if blocklist contains configured url
+ * If both allowlist and blocklist are used, return false if allowlist does not contain or if blocklist contains configured url
+ * @param {Array|string} blockedUrls
+ * @param {Array|string} allowedUrls
+ * @param {string} configuredUrls
+ * @returns {boolean} true if the configuredUrl is valid
```

Comment:
what's difference compare to `isValidURL`

Reply:
`isValidURL` checks whether if a specific IP address fall in the range of an IP address block
`checkConfigUrls` is an overall function to check configured url. If blocklist is enabled, it will call `isValidURL`. Otherwise, it uses allowlist and check whether allowlist contains configured url.

Reply:
seems isValidURL is only used for validate BlockList cases, let's be more specific about name